### DISCUSSION TITLE
Is KeyCloak support urn:ietf:params:oauth:grant-type:pre-authorized_code as grant_type?

### DISCUSSION THREAD

**Discussion by nbAmit** [keycloak/keycloak#20691] [Q&A]
We are trying to implement the OIDCVCI using KeyCloak, As per [OIDCVCI speciation](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-token-request), we need to pass `grant_type` as `urn:ietf:params:oauth:grant-type:pre-authorized_code` but I am not sure whether KeyCloak supports this grant_type or not

Does any one has any idea on this?

**Comment by y-tabata** [comment]
Keycloak does not support the pre-authorized code grant type. How about using the authorization code grant described in [OID4VCI](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-authorization-code-flow)? I think it can be realized by implementing some SPI providers such as authenticator.

**Reply by nbAmit** [reply]
Thanks for your reply, @y-tabata. Do you have any example repository or article for creating custom SPI providers?
